cmd/tetragon/main.go

```diff
@@ -322,9 +322,45 @@ func tetragonExecute() error {
 		}
 	}
 
+	// Periodically log status
+	go logStatus(ctx, obs)
+
 	return obs.Start(ctx)
 }
 
+// Periodically log current status every 1 hour. For lost or error
+// events we ratelimit statistics to 1 message per every 5mins to
+// continuously infor users that events are being lost without
+// polluting logs.
+func logStatus(ctx context.Context, obs *observer.Observer) {
+	prevLost := uint64(0)
+	prevErrors := uint64(0)
+	lostTicker := time.NewTicker(5 * time.Minute)
+	defer lostTicker.Stop()
+	logTicker := time.NewTicker(1 * time.Hour)
+	defer logTicker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
```

Comment:
Do we print the stats elsewhere when we the agent exits?

Reply:
obs.PrintStats() is inside the signal handling path.